Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -847,6 +847,48 @@ extension BrowserViewController: BrowserToolbarDelegate {
     }
 }
 
+protocol WindowCloseHelperDelegate: class {
+    func windowCloseHelper(windowCloseHelper: WindowCloseHelper, didRequestToCloseBrowser browser: Browser)
+}
+
+class WindowCloseHelper: BrowserHelper {
+    weak var delegate: WindowCloseHelperDelegate?
+    private weak var browser: Browser?
+
+    required init(browser: Browser) {
+        self.browser = browser
+        if let path = NSBundle.mainBundle().pathForResource("WindowCloseHelper", ofType: "js") {
+            if let source = NSString(contentsOfFile: path, encoding: NSUTF8StringEncoding, error: nil) as? String {
+                var userScript = WKUserScript(source: source, injectionTime: WKUserScriptInjectionTime.AtDocumentEnd, forMainFrameOnly: true)
```

Comment:
You mean change it to `AtDocumentStart` ? Also, I see now that `forMainFrameOnly:` is set to `true`. I wonder if Persona specifically runs stuff in iframe, which would mean this code is never injected! That could actually be the fix for Persona.